source/dofs/dof_handler.cc

```diff
@@ -2128,14 +2128,38 @@ void
 DoFHandler<dim, spacedim>::distribute_dofs(
   const hp::FECollection<dim, spacedim> &ff)
 {
-  Assert(
-    this->tria != nullptr,
-    ExcMessage(
-      "You need to set the Triangulation in the DoFHandler using reinit() or "
-      "in the constructor before you can distribute DoFs."));
+#ifdef DEBUG
```

Comment:
The only expensive check is the one that starts at line 2148. Would you mind moving the `#ifdef DEBUG` down to that place?